user_assigned_identity/role_assignments.tf

```diff
@@ -1,5 +1,5 @@
 resource "azurerm_role_assignment" "roles" {
-  for_each                               = var.role_assignments
+  for_each                               = { for key, value in var.role_assignments : key => value if value.scope != null }
```

Comment:
The for_each can reference var.role_assignments directly without checks. This logic is handled in ../role_assignments.tf now (for the var.identity.role_assignments).